src/libs/actions/Report.js

```diff
@@ -425,6 +417,65 @@ function removeOptimisticActions(reportID) {
     });
 }
 
+/**
+ * If an iouReport is open (has an IOU, but is not yet paid) then we maintain a link between it and the associated
+ * chatReport in Onyx, simplifying IOU data retrieval and reducing necessary API calls when displaying IOU components:
+ * - chatReport: {id: 123, iouReportID: 987, ...}
+ * - iouReport: {id: 987, chatReportID: 123, ...}
+ *
+ * This function allows us to fetch an iouReport without updating the linked chatReport, preventing the chatReport's
+ * 'iouReportID' value from being updated. As an example, this is desired when fetching historical reports, to avoid
+ * overwritting an open iouReportID with a closed iouReportID. If this was to occur, unpaid IOUs would not be
+ * highlighted to the user.
+ *
+ * If updating the chatReport's data is desired, use `fetchIOUReportByIDAndUpdateChatReportLink` instead.
+ *
+ * @param {Number} iouReportID - ID of the report we are fetching
+ * @param {Number} chatReportID - associated chatReportID, set as an iouReport field, but not used to maintain the link
+ */
+function fetchIOUReportByID(iouReportID, chatReportID) {
+    fetchIOUReport(iouReportID, chatReportID)
+        .then(iouReportObject => setLocalIOUReportData(iouReportObject));
+}
+
+/**
+ * If an iouReport is open (has an IOU, but is not yet paid) then we maintain a link between it and the associated
+ * chatReport in Onyx, simplifying IOU data retrieval and reducing necessary API calls when displaying IOU components:
+ * - chatReport: {id: 123, iouReportID: 987, ...}
+ * - iouReport: {id: 987, chatReportID: 123, ...}
+ *
+ * This link must remain in sync when the iouReport is modified. This function forces a link update after fetching the
+ * iouReport and therefore should only be called if we are certain that the fetched iouReport is currently open or about
+ * to be made open - else we would overwrite the existing open iouReport link with a closed iouReport.
+ *
+ * Examples of usage include 'receieving a push notification', or 'paying an IOU', because both of these cases can only
+ * occur for an iouReport that is currently open (notifications are not sent for closed iouReports, and you cannot pay a
+ * closed IOU).
+ *
+ * @param {Number} iouReportID - ID of the report we are fetching
+ * @param {Number} chatReportID - associated chatReportID, used to maintain the link between reports
+ */
+function fetchIOUReportByIDAndUpdateChatReportLink(iouReportID, chatReportID) {
+    fetchIOUReport(iouReportID, chatReportID)
+        .then((iouReportObject) => {
+            // First persist the IOU Report data to Onyx
+            setLocalIOUReportData(iouReportObject);
```

Comment:
Use `fetchIOUReportByID()` instead of setting the local data here?

Reply:
NAB, can keep the comment about persisting the data if you think it's relevant.

Reply:
Leaving this, as we would need to make `setLocalIOUReportData` return the object that was passed in. It seems a bit simpler to read this way.

Reply:
>we would need to make setLocalIOUReportData return the object that was passed in

Why? Just make `fetchIOUReportByID()` return the object after it calls this.

Reply:
Aren't we supposed to avoid returning from Actions? It's currently called from another Action, but this isn't guaranteed to be the case.

Reply:
Actually, it's also called from a component.

Reply:
Okay, turns out this isn't a problem after all -- implementing.